### PR TITLE
feat: Adds the DropdownContainer component

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
Adds a new component called `DropdownContainer`. The component is a horizontal container that displays overflowed items in a popover. It also displays an indicator of how many items are currently overflowing.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/198854261-985c2419-7458-47f2-b643-57206f6bbbd4.mov

### TESTING INSTRUCTIONS
Open the Storybook and check the new component.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
I already have RTL tests locally but some are still failing. I'll add them soon. I'll also add an MDX file with an overview of the component.

File: superset-frontend/src/components/DropdownContainer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      <div
+        css={css`
+          display: flex;
+          flex-direction: column;
+          gap: ${theme.gridUnit * 3}px;
+          width: 200px;
+        `}
+        style={popoverStyle}
+      >
+        {overflowedItems}
```
Comment (kgabryje) [inline superset-frontend/src/components/DropdownContainer/index.tsx:180]: Can we make the popover content customizable via props as well? In the case of horizontal filter bar we'll need to hide filters out of scope in a collapsible inside of popover

File: superset-frontend/src/components/DropdownContainer/index.tsx [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    () =>
+      overflowingIndex !== -1
+        ? items
+            .slice(overflowingIndex, items.length)
+            .map(item => React.cloneElement(item.element, { key: item.id }))
+        : [],
+    [items, overflowingIndex],
+  );
+
+  const overflowedIds = useMemo(
```
Comment (kgabryje) [inline superset-frontend/src/components/DropdownContainer/index.tsx:152]: Could we maybe merge calculating items and ids in 1 operation? Like `const [overflowedItems, overflowedIds] = useMemo(...)`

File: superset-frontend/src/components/DropdownContainer/index.tsx [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+  style,
+}: DropdownContainerProps) => {
+  const theme = useTheme();
+  const { ref, width = 0 } = useResizeDetector<HTMLDivElement>();
+  const previousWidth = usePrevious(width) || 0;
+  const { current } = ref;
+  const [overflowingIndex, setOverflowingIndex] = useState<number>(-1);
+  const [itemsWidth, setItemsWidth] = useState<number[]>([]);
+
+  useEffect(() => {
```
Comment (kgabryje) [inline superset-frontend/src/components/DropdownContainer/index.tsx:82]: I think `useLayoutEffect` is preferred when doing stuff like layout measurements https://dev.to/emmanuelthecoder/useeffect-vs-uselayouteffect-the-difference-and-when-to-use-them-124c

File: superset-frontend/src/components/DropdownContainer/index.tsx (hunk 1 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/DropdownContainer/index.tsx]: I added the `popoverContent` property 👍🏼

File: superset-frontend/src/components/DropdownContainer/index.tsx (hunk 2 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/DropdownContainer/index.tsx]: Done 👍🏼

File: superset-frontend/src/components/DropdownContainer/index.tsx (hunk 3 above)
Comment (michael-s-molina) [reply superset-frontend/src/components/DropdownContainer/index.tsx]: Done 👍🏼

**Comment by michael-s-molina** [comment]
> in the recording we can see some flashing of appearing/disappearing elements during resizing. I'm assuming that happens when container size is on the verge of fitting or not an additional element. Could we maybe add a few pixels hysteresis to changing the state of displayed/hidden elements?

Actually, this is somewhat related to Select's horizontal label. If you set the label position at the top or remove it, you'll see it's no longer flicking. I'll check this alongside the RTL tests. I just wanted to resolve the blocking ones first in case we need to merge the PR to unblock other things.

**Review by geido — APPROVED** [review]
Tested locally. LGreaTM!

**Comment by codyml** [comment]
@michael-s-molina This looks great, thanks!  Would it be possible to add a prop for customizing the value in the badge?  For horizontal filter bar, I believe we'll need to show the number of active filters, not the total number of filters.  If there was a badge count prop, we could handle calls to `onOverflowingStateChange` by checking which overflowing filters in the dropdown are active and setting the badge accordingly.

**Comment by codyml** [comment]
@michael-s-molina For the horizontal filter bar, I think we may also need a way to programmatically open/close the dropdown.  Would that be possible to include here, maybe with a `useImperativeHandle`?  I think we'll also want to be able to scroll to and highlight overflow content, so could we also maybe expose a ref to the dropdown content container?

**Comment by kgabryje** [comment]
Merged to unblock other tasks, we'll address @codyml's comments in other PRs

**Pull request merged**